src/main/java/org/openmrs/addonindex/service/ElasticSearchIndex.java

```diff
@@ -97,19 +97,36 @@ public void index(AddOnInfoAndVersions infoAndVersions) throws IOException {
 		
 		BoolQueryBuilder boolQB = QueryBuilders.boolQuery();
 		if (type != null) {
+			//Gets all modules of the type specified by variable type
```

Comment:
I understand what you mean. I'll commit the changes with the modified comments and hopefully learn from what you've said:). And to address the usgae of match vs term here , I do agree that term would've been the right choice but it doesn't actually make a difference as type is just word and hence there's nothing to be tokenized.